display/d.linegraph/main.c

```diff
@@ -89,7 +89,7 @@ static int cmp(const void *a, const void *b)
 static char *icon_files(void)
 {
     char **list, *ret;
-    char buf[GNAME_MAX], path[GPATH_MAX], path_i[GPATH_MAX];
+    char buf[GNAME_MAX + GNAME_MAX], path[GPATH_MAX], path_i[GPATH_MAX];
```

Comment:
```suggestion
    char buf[GPATH_MAX], path[GPATH_MAX], path_i[GPATH_MAX];
```
A path is stored in buf, thus GPATH_MAX seems appropriate.

Reply:
> A path is stored in buf, thus GPATH_MAX seems appropriate.

It is not a path, but a concatenated string of two directory names `[dirname]/[dirname]` like e.g. `basic/arrow`, which is exactly maximum `GNAME_MAX + GNAME_MAX`.

Reply:
Plus the `/` char, i.e. `char buf[GNAME_MAX + GNAME_MAX + 1]`?

Reply:
The dirnames input has already included a null character each, now one of them will replaced by a `/` .

Reply:
Just to clarify the comment above.

`#define GNAME_MAX 256`

"The maximum length of a name in a directory is MAXNAMLEN" ([Darwin](https://github.com/apple/darwin-xnu/blob/main/bsd/sys/dirent.h)):
```
#define MAXNAMLEN   255
struct dirent {
    ...
    char d_name[MAXNAMLEN + 1];
}
```
See also [glibc](https://sourceware.org/git/?p=glibc.git;a=blob;f=dirent/dirent.h;h=1d1fab7e55cdad4d1196e48a35d7712e4e390f92;hb=refs/heads/master).

Thus:
```
[dirname1] + '/' + [dirname2] + '\0'  =  (GNAME_MAX + GNAME_MAX)
    255       1       255         1   =  (   256    +   256    )
```